src/app/item-popup/ItemStat.tsx

```diff
@@ -176,3 +220,50 @@ function getModdedStatValue(item: DimItem, stat: DimStat) {
 export function isD1Stat(item: DimItem, _stat: DimStat): _stat is D1Stat {
   return item.isDestiny1();
 }
+
+/*
+ * Finds the sockets that have plugs which are grouped under the energy meter category
+ * The energy category is where the masterwork plug lives in armor 2.0
+ */
+function getArmor2MasterworkSockets(item: D2Item) {
+  if (!item.sockets) {
+    return [];
+  }
+
+  const masterworkSocketCategory = item.sockets.categories.find((category) => {
+    return category.category.categoryStyle === DestinySocketCategoryStyle.EnergyMeter;
+  });
+  const masterworkSocketHashes =
+    (masterworkSocketCategory && getPlugHashesFromCategory(masterworkSocketCategory)) || [];
+
+  return item.sockets.sockets.filter((socket) => {
+    const plugHash = idx(socket, (socket) => socket.plug.plugItem.hash) || null;
+    return masterworkSocketHashes.includes(plugHash);
+  });
+}
+
+/*
+ * Sums up all the armor statistics from the plug in the socket
+ */
+function getSumOfArmorStats(sockets: DimSocket[]) {
+  return _.sum(
+    _.map(sockets, (socket) => {
```

Comment:
Prefer `sockets.map` instead of `_.map(sockets`. Or better yet, `_.sumBy(sockets, (socket) => ...)` which won't create intermediate array garbage.